diff -r 33245988e8a2 src/file/SDL_rwops.c
--- a/src/file/SDL_rwops.c	Sun Sep 04 14:57:10 2011 -0700
+++ b/src/file/SDL_rwops.c	Mon Sep 05 22:36:56 2011 -0600
@@ -445,7 +445,7 @@
         SDL_SetError("SDL_RWFromFile(): No file or no mode specified");
         return NULL;
     }
-#if defined(ANDROID)
+#if defined(ANDROID) && defined(DONT_CARE_WAFFLES)
     rwops = SDL_AllocRW();
     if (!rwops)
         return NULL;            /* SDL_SetError already setup by SDL_AllocRW() */
diff -r 33245988e8a2 src/joystick/SDL_joystick.c
--- a/src/joystick/SDL_joystick.c	Sun Sep 04 14:57:10 2011 -0700
+++ b/src/joystick/SDL_joystick.c	Mon Sep 05 22:36:56 2011 -0600
@@ -26,7 +26,6 @@
 #include "SDL_sysjoystick.h"
 #include "SDL_joystick_c.h"
 #include "SDL_assert.h"
-
 #if !SDL_EVENTS_DISABLED
 #include "../events/SDL_events_c.h"
 #endif
diff -r 33245988e8a2 src/joystick/android/SDL_sysjoystick.c
--- a/src/joystick/android/SDL_sysjoystick.c	Sun Sep 04 14:57:10 2011 -0700
+++ b/src/joystick/android/SDL_sysjoystick.c	Mon Sep 05 22:36:56 2011 -0600
@@ -88,6 +88,7 @@
     int i;
     float values[3];
 
+    return;
     Android_JNI_GetAccelerometerValues(values);
 
     for ( i = 0; i < 3; i++ ) {
diff -r 33245988e8a2 src/render/SDL_render.c
--- a/src/render/SDL_render.c	Sun Sep 04 14:57:10 2011 -0700
+++ b/src/render/SDL_render.c	Mon Sep 05 22:36:56 2011 -0600
@@ -52,12 +52,12 @@
 #if SDL_VIDEO_RENDER_OGL
     &GL_RenderDriver,
 #endif
+#if SDL_VIDEO_RENDER_OGL_ES
+    &GLES_RenderDriver,
+#endif
 #if SDL_VIDEO_RENDER_OGL_ES2
     &GLES2_RenderDriver,
 #endif
-#if SDL_VIDEO_RENDER_OGL_ES
-    &GLES_RenderDriver,
-#endif
 #if SDL_VIDEO_RENDER_DIRECTFB
     &DirectFB_RenderDriver,
 #endif
